src/pages/Vote/VotingWallet/index.tsx

```diff
@@ -64,20 +139,86 @@ export const VoteUi: React.FC<IVoteUiProps> = ({ votingWeight, xvsLockedWei }) =
           <Trans
             i18nKey="vote.delegateYourVoting"
             components={{
-              Anchor: <span css={styles.clickableText} role="button" aria-pressed="false" />,
+              Anchor: (
+                <span
+                  css={styles.clickableText}
+                  role="button"
+                  aria-pressed="false"
+                  tabIndex={0}
+                  onClick={() => setDelegateModelIsOpen(true)}
+                />
+              ),
             }}
           />
         </Typography>
       </Paper>
+      <DelegateModal
+        onClose={() => setDelegateModelIsOpen(false)}
+        isOpen={delegateModelIsOpen}
+        currentUserAccountAddress={currentUserAccountAddress}
+        previouslyDelegated={previouslyDelegated}
+        setVoteDelegation={setVoteDelegation}
+        isVoteDelegationLoading={isVoteDelegationLoading}
+      />
     </div>
   );
 };
 
-const Vote: React.FC = () => (
-  <VoteUi
-    votingWeight={new BigNumber(1.003)}
-    xvsLockedWei={new BigNumber('19931200345567000000')}
-  />
-);
+const VotingWallet: React.FC = () => {
+  const [delegateModelIsOpen, setDelegateModelIsOpen] = useState(false);
+  const { t } = useTranslation();
+  const { account: { address: accountAddress } = { address: undefined }, openAuthModal } =
+    useContext(AuthContext);
+
+  const { data: currentVotesWei } = useGetCurrentVotes(
+    { accountAddress: accountAddress || '' },
+    { enabled: !!accountAddress },
+  );
+  const { data: delegate } = useGetVoteDelegateAddress(
+    { accountAddress: accountAddress || '' },
+    { enabled: !!accountAddress },
+  );
+
+  const { data: vaults } = useGetVestingVaults({ accountAddress });
+  const [xvsVault] = vaults.filter(v => v.stakedTokenId === XVS_TOKEN_ID);
+  const userStakedWei = xvsVault?.userStakedWei || new BigNumber(0);
+
+  const { openSuccessfulTransactionModal } = useSuccessfulTransactionModal();
+  const { mutateAsync: setVoteDelegation, isLoading: isVoteDelegationLoading } = useSetVoteDelegate(
+    {
+      onSettled: data => {
```

Comment:
It's the `onSuccess` that needs to be used here, otherwise this callbacks will also gets executed on failure in which case `data` would correspond to the error.

We also still need to add an `onError` callback to display a toast on failure.